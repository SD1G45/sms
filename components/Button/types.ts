export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  invert?: boolean;
  disabled?: boolean;
  className?: string;
  loading?: boolean;
}
