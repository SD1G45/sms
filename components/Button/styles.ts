import React from "react";
import styled from "styled-components";
import { ButtonProps } from "./types";

export const StyledButton = styled.button<ButtonProps>`
  width: 100%;
  padding: 12px 14px;
  background-color: ${(props) => (props.invert ? "white" : "#4881F0")};
  border-radius: 18px;
  border: 2px solid ${(props) => (props.invert ? "#4881F0" : "white")};
  color: ${(props) => (props.invert ? "#4881F0" : "white")};
  cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};
  font-size: 1rem;
  opacity: ${(props) => (props.disabled ? "40%" : "100%")};
  overflow: hidden;
  white-space: nowrap;

  &:hover {
    opacity: ${(props) => (props.disabled ? "40%" : "70%")};
  }
`;

export const LoadingContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  gap: 10px;
`;
