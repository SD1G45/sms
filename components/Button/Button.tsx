import React from "react";
import { StyledButton } from "./styles";
import { ButtonProps } from "./types";

const Button: React.FC<ButtonProps> = ({
  invert,
  className,
  children,
  ...other
}) => {
  return (
    <StyledButton invert={invert} className={className} {...other}>
      {children}
    </StyledButton>
  );
};

export default Button;
