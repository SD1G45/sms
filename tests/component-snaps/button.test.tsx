import React from 'react';
import renderer from "react-test-renderer";
import "jest-styled-components";
import Button from "../../components/Button";

test('Button renders correctly', () => {
  const tree = renderer.create(<Button/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders as disabled', () => {
  const tree = renderer.create(<Button disabled={true}/>).toJSON();
  expect(tree).toMatchSnapshot();
}); 

test('Button renders as loading', () => {
  const tree = renderer.create(<Button loading={true}/>).toJSON();
  expect(tree).toMatchSnapshot();
});