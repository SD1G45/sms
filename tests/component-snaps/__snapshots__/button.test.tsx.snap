// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders as disabled 1`] = `
.c0 {
  width: 100%;
  padding: 12px 14px;
  background-color: #4881F0;
  border-radius: 18px;
  border: 2px solid white;
  color: white;
  cursor: not-allowed;
  font-size: 1rem;
  opacity: 40%;
  overflow: hidden;
  white-space: nowrap;
}

.c0:hover {
  opacity: 40%;
}

<button
  className="c0"
  disabled={true}
/>
`;

exports[`Button renders as loading 1`] = `
.c2 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  width: 20px;
  height: 20px;
}

.c3 {
  box-sizing: border-box;
  display: block;
  width: 20px;
  height: 20px;
  border: 3px solid #fff;
  border-radius: 50%;
  -webkit-animation: fEWCgj 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: fEWCgj 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border-color: #fff transparent transparent transparent;
}

.c3:nth-child(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.c3:nth-child(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c3:nth-child(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

.c0 {
  width: 100%;
  padding: 12px 14px;
  background-color: #4881F0;
  border-radius: 18px;
  border: 2px solid white;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  opacity: 100%;
  overflow: hidden;
  white-space: nowrap;
}

.c0:hover {
  opacity: 70%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 10px;
}

<button
  className="c0"
  loading={true}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      color="#fff"
      size={20}
    >
      <div
        className="c3"
        color="#fff"
        size={20}
      />
    </div>
    <div>
      Loading
    </div>
  </div>
</button>
`;

exports[`Button renders correctly 1`] = `
.c0 {
  width: 100%;
  padding: 12px 14px;
  background-color: #4881F0;
  border-radius: 18px;
  border: 2px solid white;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  opacity: 100%;
  overflow: hidden;
  white-space: nowrap;
}

.c0:hover {
  opacity: 70%;
}

<button
  className="c0"
/>
`;
